local/core/sdkError.go

```diff
@@ -0,0 +1,141 @@
+package core
+
+import (
+	"fmt"
+
+	"google.golang.org/grpc/codes"
+)
+
+type SdkError struct {
```

Comment:
SdkError can wrap another SdkError - this need to be solved - if it happens we just return err instead of SdkError.

Reply:
Any suggestions on how to solve this issue?

Reply:
when you eg `MakeUnavailable` check err if it contains SdkError and if it is true return original err.